include/librealsense2/h/rs_option.h

```diff
@@ -246,6 +246,15 @@ extern "C" {
     */
     int rs2_is_option_read_only(const rs2_options* options, rs2_option option, rs2_error** error);
 
+    /**
+     * check if an option is volatile, that is, value set by us can change without us setting it again
+     * \param[in] options  the options container
+     * \param[in] option   option id to be checked
+     * \param[out] error   if non-null, receives any error that occurs during this call, otherwise, errors are ignored
+     * \return true if option is volatile
+     */
+    int rs2_is_option_volatile( const rs2_options * options, rs2_option option, rs2_error ** error );
```

Comment:
The major concern is if we add this API and we have many options that are `volatile` and can be changed by other, it will return false on them.
Even when they can be changed..
Thoughts?

Reply:
This is an addition for future product, currently we didn't have the need so all options are "not volatile".

I looked at the options, didn't see many that might be changed to "volatile". Maybe "inter cam sync mode" or "thermal compensation" that are more HW/FW related. If there will be a need we can always mark some current options as "volatile".

We can present this to the SW architect and ask his opinion if other options can benefit from this addition.